streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/CombinedKey.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals.foreignkeyjoin;
+
+public class CombinedKey<KF, KP> {
+    private final KF foreignKey;
+    private KP primaryKey = null;
+
+    public CombinedKey(final KF foreignKey, final KP primaryKey) {
+        this.foreignKey = foreignKey;
+        this.primaryKey = primaryKey;
+    }
+
+    public CombinedKey(final KF foreignKey) {
+        this.foreignKey = foreignKey;
+    }
+
+    public KF getForeignKey() {
```

Comment:
```suggestion
    KF getForeignKey() {
```

In general, we try to retrict visibility to the minimum required. I won't call these out further at this time.